Lib/test/test_math.py

```diff
@@ -1992,6 +1992,16 @@ def test_ulp(self):
             with self.subTest(x=x):
                 self.assertEqual(math.ulp(-x), math.ulp(x))
 
+    def test_issue39871(self):
+        # A SystemError should not be raised if the first arg to atan2(),
+        # copysign(), or remainder() cannot be converted to a float.
+        class F:
+            def __float__(self):
+                1/0
+        for func in math.atan2, math.copysign, math.remainder:
+            with self.assertRaises(TypeError):
+                func('', F())
```

Comment:
Here the purpose of ``''`` is not obvious neither. Maybe explain that an empty string cannot be casted to float.

```suggestion
                func('str cannot be converted to float', F())
```

float() does accept str which confused me :-)

```
>>> float('1.0')
1.0
```

Reply:
There is nothing special about *empty* string. You can use any object which cannot be converted to float implicitly.
```suggestion
                func('non-number', F())
```